test/async_progress_queue_worker.js

```diff
@@ -4,10 +4,144 @@ const common = require('./common');
 const assert = require('assert');
 
 module.exports = common.runTest(test);
+const nodeVersion = process.versions.node.split('.')[0];
+
+let asyncHooks;
+function checkAsyncHooks () {
+  if (nodeVersion >= 8) {
+    if (asyncHooks === undefined) {
+      asyncHooks = require('async_hooks');
+    }
+    return true;
+  }
+  return false;
+}
 
 async function test ({ asyncprogressqueueworker }) {
   await success(asyncprogressqueueworker);
   await fail(asyncprogressqueueworker);
+
+  await asyncProgressWorkerCallbackOverloads(asyncprogressqueueworker.runWorkerWithCb);
+  await asyncProgressWorkerRecvOverloads(asyncprogressqueueworker.runWorkerWithRecv);
+  await asyncProgressWorkerNoCbOverloads(asyncprogressqueueworker.runWorkerNoCb);
+}
+
+async function asyncProgressWorkerCallbackOverloads (bindingFunction) {
+  bindingFunction(common.mustCall());
+  if (!checkAsyncHooks()) {
+    return;
+  }
+
+  const hooks = common.installAysncHooks('cbResources');
+
+  const triggerAsyncId = asyncHooks.executionAsyncId();
+  await new Promise((resolve, reject) => {
+    bindingFunction(common.mustCall(), 'cbResources');
+    hooks.then(actual => {
+      assert.deepStrictEqual(actual, [
+        {
+          eventName: 'init',
+          type: 'cbResources',
+          triggerAsyncId: triggerAsyncId,
+          resource: {}
+        },
+        { eventName: 'before' },
+        { eventName: 'after' },
+        { eventName: 'destroy' }
+      ]);
+    }).catch(common.mustNotCall());
+    resolve();
```

Comment:
The `common.mustNotCall()` should be the Promise's `reject` callback, and the `resolve()` belongs inside the `hooks.then()` callback after the assertion. 

This will provide the actual assertion error (in case of failure). Try changing the assertion object, eg. the `destroy` event name to `__destroy`.

With your code, you'll get:

```text
> AssertionError [ERR_ASSERTION]: function should not have been called
    at mustNotCall (/Users/kevineady/Documents/Projects/node-addon-api/test/common/index.js:120:12) {
  generatedMessage: false,
  code: 'ERR_ASSERTION',
  actual: undefined,
  expected: undefined,
  operator: 'fail'
}
```

With the suggested changes, you'll get:

```text
AssertionError [ERR_ASSERTION]: Expected values to be strictly deep-equal:
+ actual - expected ... Lines skipped

  [
    {
...
    },
    {
+     eventName: 'destroy'
-     eventName: '__destroy'
    }
  ]
    at /Users/kevineady/Documents/Projects/node-addon-api/test/async_progress_queue_worker.js:41:14 {
  generatedMessage: true,
  code: 'ERR_ASSERTION',
  actual: [Array],
  expected: [Array],
  operator: 'deepStrictEqual'
}
```

Reply:
Sorry for the wait. Thank you for pointing this out! I borrowed this pattern from other places in the test suites. Will draft up another PR to make these fixes.